pydra/engine/helpers_file.py

```diff
@@ -70,22 +70,20 @@ def hash_file(afile, chunk_len=8192, crypto=sha256, raise_notfound=True):
     """Compute hash of a file using 'crypto' module."""
     from .specs import LazyField
 
-    if afile is None or isinstance(afile, LazyField) or isinstance(afile, list):
-        return None
-    if not Path(afile).is_file():
-        if raise_notfound:
-            raise RuntimeError('File "%s" not found.' % afile)
+    try:
+        crypto_obj = crypto()
+        with open(afile, "rb") as fp:
+            while True:
+                data = fp.read(chunk_len)
+                if not data:
+                    break
+                crypto_obj.update(data)
+        return crypto_obj.hexdigest()
+    except:
+        if not Path(afile).is_symlink() and raise_notfound:
+            raise FileNotFoundError(f"File {afile} not found.")
         return None
```

Comment:
I would suggest the following:

```suggestion
    try:
        fp = open(afile, "rb")
    except FileNotFoundError:
        if not Path(afile).is_symlink() and raise_notfound:
            raise
        return None    
    crypto_obj = crypto()
    with fp as fp:
        while True:
            data = fp.read(chunk_len)
            if not data:
                break
            crypto_obj.update(data)
    return crypto_obj.hexdigest()
    
```

This keeps the potentially-failing block inside `try:` to a minimum. Using `with` on the resulting object will still have the nice cleanup properties. I also just re-raise the original error, which presumably should have the same information.

Reply:
@nicolocin - will you be addressing @effigies review?

Reply:
@effigies @satra - so sorry thought I replied!  Yes this looks better!  Thanks a lot